### ISSUE TITLE
`WildcardMatcher` without a wildcard character matches strings that have the same prefix

### ISSUE THREAD

**Issue by ColmHally** [DamonOehlman/wildcard#5]
If I create a `WildcardMatcher` with an asterisk, it matches strings that have a common prefix as the pattern used to create the matcher:
```javascript
var WildcardMatcher = require('wildcard')
var wildcard = new WildcardMatcher("test")

wildcard.match("test") // returns a match as expected
wildcard.match("not.a.match") // returns false as expected

wildcard.match("test.second.level.should.not.match") // unexpectedly returns a match
```

Since I have not specified a string that includes a wildcard character, I strongly believe there is a defect here and I will upload a PR shortly with a fix for this.

**Comment by DamonOehlman** [comment]
Hey Colm - thanks for reporting this and a big thanks to @danielkrainas for fixing the issue.  The code has been merged and version `1.1.1` of the package published.

**Comment by ColmHally** [comment]
Great, thanks @DamonOehlman @danielkrainas !
